### PR TITLE
Add TestRegulatedTokens, migration, and mint script.

### PR DESCRIPTION
**Pull request by johnsetzer**

### CODE REVIEW

File: test/TestRegulatorService.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  let regulator;
+  let token;
+
+  beforeEach(async () => {
+    regulator = await TestRegulatorService.new({ from: owner });
+    const registry = await ServiceRegistry.new(regulator.address);
+    token = await TestRegulatedToken.new(registry.address, 'Test', 'TEST');
+  });
+
+  const assertBalances = async balances => {
```
Comment (foobarfighter) [inline test/TestRegulatorService.js]: Can you create a helper for this so we aren't duplicating code?

File: TEST_TOKENS.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# TestRegulatedTokens
```
Comment (foobarfighter) [inline TEST_TOKENS.md]: Please move this to the README

File: contracts/TestRegulatorService.sol [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ * It returns check() reason codes based on the amount of the transaction.
+ */
+contract TestRegulatorService is RegulatorService  {
+  uint8 constant private SUCCESS = 0;  /* 0 <= AMOUNT < 10 */
+  uint8 constant private ELOCKED = 1;  /* 10 <= AMOUNT < 20 */
+  uint8 constant private EDIVIS = 2;   /* 20 <= AMOUNT < 30 */
+  uint8 constant private ESEND = 3;    /* 30 <= AMOUNT < 40 */
+  uint8 constant private ERECV = 4;    /* 40 <= AMOUNT < 50 */
+
+  function check(address _token, address _spender, address _from, address _to, uint256 _amount) public returns (uint8) {
```
Comment (foobarfighter) [inline contracts/TestRegulatorService.sol]: Please add documentation

File: truffle.js [hunk 4]
```diff
@@ -1,9 +1,29 @@
+const Web3 = require('web3');
+const web3 = new Web3();
+const HDWalletProvider = require('truffle-hdwallet-provider');
+
+let infuraTestNetworks = {}
+var kovanMnemonic = process.env['RTOKEN_KOVAN_MNEMONIC'];
+if (kovanMnemonic) {
```
Comment (foobarfighter) [inline truffle.js]: We can get rid of this conditional

File: TEST_TOKENS.md (hunk 2 above)
Comment (johnsetzer) [reply TEST_TOKENS.md]: Done

File: truffle.js (hunk 4 above)
Comment (johnsetzer) [reply truffle.js]: Done

File: test/TestRegulatorService.js (hunk 1 above)
Comment (johnsetzer) [reply test/TestRegulatorService.js]: I agree that assertBalances() and checkResult() are annoying duplicated code. A lot of their value comes from having closures on local variables. The only ways I can think of to reduce boiler plate are to pass in more parameters to the functions or to create a Factory method that returns an instance of assertBalances() with the needed closures. Both those options seem like more trouble than they are worth. I think we are best off leaving for now and refactoring if we see ourselves doing this more.

I'm open to suggestions.

**Pull request merged**